### PR TITLE
Epmcuxda-6570: Error handlers and validation

### PR DESCRIPTION
**Pull request by slabkodima90**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -10,16 +10,15 @@
     "allowSyntheticDefaultImports": true,
     "allowJs": false,
     "resolveJsonModule": true,
-    "noImplicitAny": true,
```
Comment (mishelen) [inline tsconfig.json]: Could you try without `any` ;) considers that such things have to be discussed separately I mean changing code style

File: .prettierrc.js [hunk 2]
```diff
@@ -5,7 +5,6 @@ module.exports = {
   bracketSpacing: true,
   printWidth: 120,
   tabWidth: 2,
-  printWidth: 120,
```
Comment (mishelen) [inline .prettierrc.js]: The same thing, it's not related to PR

File: src/exceptions/middleware-error-handler.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { Request, Response, NextFunction } from 'express';
```
Comment (mishelen) [inline src/exceptions/middleware-error-handler.ts]: I have questions regarding folder and file names. Each function you pass to express or routers is middleware. For me this folder only appropriate for storing error classes.

My opinion is to place this file as error controller but without extending Abstract controller, then directly launch it in app.

File: src/exceptions/process-error-handler.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { Logger } from 'winston';
```
Comment (mishelen) [inline src/exceptions/process-error-handler.ts]: Great, appreciate it!
But could you move it in index file to explicitly handle all top level errors?

**Comment by mishelen** [comment]
One more note regarding your commit style. As I see both you commits can be related to both goals of this PR. Even commits titles and descriptions are the same. Please try distinguish object of changes and dont write common messages

File: tsconfig.json (hunk 1 above)
Comment (slabkodima90) [reply tsconfig.json]: Yeah, but we've already had such option in config.

File: src/index.ts [hunk 5]
```diff
@@ -7,6 +7,16 @@ import createApp from './app';
 import connectDB from './db';
 import { logger } from './configs/logger';
 
+process.on('uncaughtException', (error: Error) => {
```
Comment (mishelen) [inline src/index.ts]: move on bottom on file, under hood all node.js listeners are added in `process.nextTick` so it no difference where you place such  things (ofcorse  taking in account lexical scope)

**Pull request merged**